### PR TITLE
[700] Switch IEditingContext#id and IRepresentation#id from UUID to String

### PR DESCRIPTION
**Pull request by hmarchadour**
### Type of this PR 

- [ ] Bug fix
- [x] New feature or improvement
- [ ] Documentation
- [ ] Cleanup
- [ ] Test
- [ ] Build/Releng

### Issue(s)

#700

### What does this PR do?

This PR only changes the internal behavior. The type of editing context id and representation id is not constraint to a UUID anymore.
We keep using UUID in string format within sirius-web but other products will be able to use their own id policy for editing context id and representation id.

### How to test this PR?

All existing tests should be OK

### Checklist

- [ ] I have read CONTRIBUTING carefully.
- [ ] I have signed the [Contributor License Agreement](http://www.eclipse.org/legal/CLA.php).
- [ ] All my commits are signed-off (`-s`) with my mail address of my Eclipse Account.
- [ ] I have covered my changes by unit tests or integration tests or manual tests.
- [ ] All tests pass.
- [ ] I have updated the documentation accordingly
- [ ] New React components are availables in storybook

### CODE REVIEW

File: backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/IncrementalLayoutDiagramConverter.java
Comment (sbegaudeau) [reply backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/IncrementalLayoutDiagramConverter.java:50]: It doesn't *have* to change but we could change it later

File: backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/ViewCreationRequest.java
Comment (sbegaudeau) [inline backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/ViewCreationRequest.java:110]: Why does that have to change in this PR? On top of that, we already have view creation request support where the parent is the diagram and the value is not null. See `CreateViewOperationHandler#handle` for example. `null` is not an accepted value here on purpose, don't change that.

File: backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/DiagramComponent.java
Comment (sbegaudeau) [inline backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/DiagramComponent.java:71]: Why are you removing that?

File: backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/DiagramComponent.java
Comment (sbegaudeau) [inline backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/DiagramComponent.java:51]: I don't mind that change but shouldn't we have instead a way for developers to provide us with their fallback identification scheme?

File: backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/NodeComponent.java
Comment (sbegaudeau) [inline backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/NodeComponent.java:271]: The identifier of a node should stay with the same behavior. The line of code that you have changed is one of the most important part of the refresh of the diagram. It is the cornerstone of the identity of a node and it *NEEDS* the identifier of its parent. The decisions of the `ADR-007` should not be modified without a very careful discussion involving at least @pcdavid and I and I don't see why changing the type of the diagram identifier should change the way the node identifier is created.

File: backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/NodeComponentProps.java
Comment (sbegaudeau) [inline backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/NodeComponentProps.java:163]: No, that should stay required on purpose, see `ADR-007`

File: backend/sirius-web-diagrams/src/test/java/org/eclipse/sirius/web/diagrams/renderer/UnsynchronizedDiagramTests.java
Comment (sbegaudeau) [inline backend/sirius-web-diagrams/src/test/java/org/eclipse/sirius/web/diagrams/renderer/UnsynchronizedDiagramTests.java:95]: Nothing should change in this file

File: backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/compatibility/modeloperations/CreateViewOperationHandler.java
Comment (sbegaudeau) [inline backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/compatibility/modeloperations/CreateViewOperationHandler.java:72]: This should stay the way it is

File: backend/sirius-web-spring-collaborative-diagrams/src/main/resources/schema/diagram.graphqls
Comment (sbegaudeau) [inline backend/sirius-web-spring-collaborative-diagrams/src/main/resources/schema/diagram.graphqls:273]: Do not change that, that is not related to the issue at hand

File: backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/handlers/InvokeNodeToolOnDiagramEventHandler.java
Comment (sbegaudeau) [inline backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/handlers/InvokeNodeToolOnDiagramEventHandler.java:109]: This file should not be *that* impacted by the change from UUID to String

File: backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/dto/UpdateNodeBoundsInput.java
Comment (sbegaudeau) [inline backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/dto/UpdateNodeBoundsInput.java:38]: I don't mind this change but this API break is a bit outside of the scope of the issue

File: backend/sirius-web-spring-collaborative-diagrams/src/main/resources/schema/diagram.graphqls
Comment (sbegaudeau) [inline backend/sirius-web-spring-collaborative-diagrams/src/main/resources/schema/diagram.graphqls:323]: you are both modifying `UpdateNodeBoundsInput#diagramElementId` and `UpdateNodePositionInput#diagramElementId` while not related to this issue but you fail to change both of them in this file so it doesn't work anymore

Comment (sbegaudeau) [reply backend/sirius-web-spring-collaborative-diagrams/src/main/resources/schema/diagram.graphqls:323]: This issue was not supposed to have any impact on our GraphQL API or frontend

File: doc/adrs/032_relax_our_constraints_on_some_of_our_identifiers.adoc
Comment (sbegaudeau) [inline doc/adrs/032_relax_our_constraints_on_some_of_our_identifiers.adoc:36]: we could also change the type of `Node#id` and `Edge#id` on the backend and not have all those impact

File: frontend/src/diagram/sprotty/WebSocketDiagramServer.tsx
Comment (sbegaudeau) [inline frontend/src/diagram/sprotty/WebSocketDiagramServer.tsx:240]: The behavior of the frontend should stay the same, this change is related to an internal API of the backend

File: frontend/src/diagram/DiagramWebSocketContainer.tsx
Comment (sbegaudeau) [inline frontend/src/diagram/DiagramWebSocketContainer.tsx:616]: The behavior of the frontend should stay the same, this change is related to an internal API of the backend

File: frontend/src/diagram/DiagramWebSocketContainer.tsx
Comment (sbegaudeau) [inline frontend/src/diagram/DiagramWebSocketContainer.tsx:484]: The behavior of the frontend should stay the same, this change is related to an internal API of the backend

**Review by sbegaudeau — CHANGES_REQUESTED** [review]
A `CHANGELOG.adoc` entry is necessary for such an API break too

File: backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/ViewCreationRequest.java
Comment (gcoutable) [reply backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/ViewCreationRequest.java:110]: Done

File: backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/DiagramComponent.java
Comment (gcoutable) [reply backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/DiagramComponent.java:71]: Done

File: backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/NodeComponent.java
Comment (gcoutable) [reply backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/NodeComponent.java:271]: Done

File: backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/NodeComponentProps.java
Comment (gcoutable) [reply backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/NodeComponentProps.java:163]: Done

File: backend/sirius-web-diagrams/src/test/java/org/eclipse/sirius/web/diagrams/renderer/UnsynchronizedDiagramTests.java
Comment (gcoutable) [reply backend/sirius-web-diagrams/src/test/java/org/eclipse/sirius/web/diagrams/renderer/UnsynchronizedDiagramTests.java:95]: Done

File: backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/compatibility/modeloperations/CreateViewOperationHandler.java
Comment (gcoutable) [reply backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/compatibility/modeloperations/CreateViewOperationHandler.java:72]: Done

File: backend/sirius-web-spring-collaborative-diagrams/src/main/resources/schema/diagram.graphqls
Comment (gcoutable) [reply backend/sirius-web-spring-collaborative-diagrams/src/main/resources/schema/diagram.graphqls:273]: Done

File: backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/handlers/InvokeNodeToolOnDiagramEventHandler.java
Comment (gcoutable) [reply backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/handlers/InvokeNodeToolOnDiagramEventHandler.java:109]: Done

File: backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/dto/UpdateNodeBoundsInput.java
Comment (gcoutable) [reply backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/dto/UpdateNodeBoundsInput.java:38]: Done

File: backend/sirius-web-spring-collaborative-diagrams/src/main/resources/schema/diagram.graphqls
Comment (gcoutable) [reply backend/sirius-web-spring-collaborative-diagrams/src/main/resources/schema/diagram.graphqls:323]: Done

File: doc/adrs/032_relax_our_constraints_on_some_of_our_identifiers.adoc
Comment (gcoutable) [reply doc/adrs/032_relax_our_constraints_on_some_of_our_identifiers.adoc:36]: Rewritten with the fact that `Node#id`, `Edge#id` and `LayoutData#id` and `Label#id` is now a `String`.

File: frontend/src/diagram/sprotty/WebSocketDiagramServer.tsx
Comment (gcoutable) [reply frontend/src/diagram/sprotty/WebSocketDiagramServer.tsx:240]: Done

File: backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/ViewCreationRequest.java
Comment (sbegaudeau) [inline backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/ViewCreationRequest.java:40]: This can be removed, since the id is a string, null is not a valid value and technically, the code does not allow null

File: backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/ViewCreationRequest.java
Comment (sbegaudeau) [inline backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/ViewCreationRequest.java:42]: Same comment

File: backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/handlers/InvokeNodeToolOnDiagramEventHandler.java
Comment (sbegaudeau) [inline backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/handlers/InvokeNodeToolOnDiagramEventHandler.java:132]: We could probably log that

**Review by sbegaudeau — CHANGES_REQUESTED** [review]
This looks good, I still want to run this before I give my approval to this PR but it should be merged very soon (probably today).

File: backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/ViewCreationRequest.java
Comment (gcoutable) [reply backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/ViewCreationRequest.java:40]: Done

File: backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/ViewCreationRequest.java
Comment (gcoutable) [reply backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/ViewCreationRequest.java:42]: Done

File: backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/handlers/InvokeNodeToolOnDiagramEventHandler.java
Comment (gcoutable) [reply backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/handlers/InvokeNodeToolOnDiagramEventHandler.java:132]: Done

**Review by sbegaudeau — APPROVED** [review]

**Pull request merged**
